src/sagemaker/jumpstart/model.py

```diff
@@ -326,10 +339,27 @@ def _is_valid_model_id_hook():
         self.region = model_init_kwargs.region
         self.sagemaker_session = model_init_kwargs.sagemaker_session
 
+        if self.model_type == JumpStartModelType.PROPRIETARY:
+            self.log_subscription_warning()
+
         super(JumpStartModel, self).__init__(**model_init_kwargs.to_kwargs_dict())
 
         self.model_package_arn = model_init_kwargs.model_package_arn
 
+    def log_subscription_warning(self) -> None:
+        """Logs customer facing message for subscribe to the proprietary model."""
+        subscription_link = verify_model_region_and_return_specs(
+            region=self.region,
+            model_id=self.model_id,
+            version=self.model_version,
+            model_type=self.model_type,
+            scope=JumpStartScriptScope.INFERENCE,
+            sagemaker_session=self.sagemaker_session,
+        ).model_subscription_link
+        JUMPSTART_LOGGER.warning(
+            get_proprietary_model_subscription_msg(self.model_id, subscription_link)
+        )
```

Comment:
Why is that a warning? Do we know at this point that the customer has _not_ subscribed?

Ideally we would wrap `create_model` in a try/catch and add this information there.

If you think that's too complex and can be a future DevX improvement, I would suggest to use an `info` log here.

Reply:
This warning message is very similar to the [wildcard warning message](https://github.com/aws/sagemaker-python-sdk/blob/615a8adcbdeba0877c00123445c014f380eb2015/src/sagemaker/jumpstart/exceptions.py#L49) we display to the customer. The idea is to give customer a message that gets printed by default of the subscription link. 

At this point the customer may or may not have subscribed to the model, but we want to show the link regardless, similar to Monarch flow. Once we have user entitlement check in our control plan I think we can do the try/catch and only show this link if the entitlement check failed.